pytorch_lightning/callbacks/batch_size_finder.py

```diff
@@ -0,0 +1,476 @@
+            loop = trainer.fit_loop
+            loop.max_steps = self._dumped_params["max_steps"]
+            trainer.limit_val_batches = self._dumped_params["limit_val_batches"]
+        elif trainer.state.fn == TrainerFn.VALIDATING:
+            loop = trainer.validate_loop
+            trainer.limit_val_batches = self._dumped_params["limit_val_batches"]
+        elif trainer.state.fn == TrainerFn.TESTING:
+            loop = trainer.test_loop
+            trainer.limit_test_batches = self._dumped_params["limit_test_batches"]
+        elif trainer.state.fn == TrainerFn.PREDICTING:
+            loop = trainer.predict_loop
+            trainer.limit_predict_batches = self._dumped_params["limit_predict_batches"]
+
+        loop.load_state_dict(deepcopy(self._dumped_params["loop_state_dict"]), force_load_progress=True)
+        if "loop_verbose" in self._dumped_params:
+            loop.verbose = self._dumped_params["loop_verbose"]
+
+    def pre_early_exit(self, trainer):
+        if trainer.fast_dev_run:
+            return
+
+        # this is required to stop the respective loops
+        if trainer.state.fn == TrainerFn.FITTING:
+            self._dumped_params["num_training_batches"] = trainer.num_training_batches
+            trainer.num_training_batches = 0
+        elif trainer.state.fn == TrainerFn.VALIDATING:
+            self._dumped_params["num_val_batches"] = trainer.num_val_batches
+            trainer.num_val_batches = [0]
+        elif trainer.state.fn == TrainerFn.TESTING:
+            self._dumped_params["num_test_batches"] = trainer.num_test_batches
+            trainer.num_test_batches = [0]
+        elif trainer.state.fn == TrainerFn.PREDICTING:
+            self._dumped_params["num_predict_batches"] = trainer.num_predict_batches
+            trainer.num_predict_batches = [0]
+
+    def post_early_exit(self, trainer):
+        if trainer.fast_dev_run:
+            return
+
+        # restore the state used to stop the respective loop
+        if trainer.state.fn == TrainerFn.FITTING:
+            trainer.num_training_batches = self._dumped_params["num_training_batches"]
+            loop = trainer.fit_loop
+        if trainer.state.fn == TrainerFn.VALIDATING:
+            trainer.num_val_batches = self._dumped_params["num_val_batches"]
+            loop = trainer.validate_loop
+        if trainer.state.fn == TrainerFn.TESTING:
+            trainer.num_test_batches = self._dumped_params["num_test_batches"]
+            loop = trainer.test_loop
+        if trainer.state.fn == TrainerFn.PREDICTING:
+            trainer.num_predict_batches = self._dumped_params["num_predict_batches"]
+            loop = trainer.predict_loop
+
+        loop.load_state_dict(self._dumped_params["loop_state_dict"], force_load_progress=True)
+        trainer.callbacks = [cb for cb in trainer.callbacks if not isinstance(cb, BatchSizeFinder)]
+
+    def on_fit_start(self, trainer, pl_module):
```

Comment:
Let's merge the duplicated code there.

And would on_fit_start + on_validation_start trigger 2 tuning ? It is an expected behaviour ?

Reply:
on_validation_start triggers tuning only when it's a `.validate` call.